app/main/views/services.py

```diff
@@ -105,10 +106,77 @@ def edit(service_id, section):
     return render_template("edit_section.html", **template_data)
 
 
+@main.route(
+    '/services/compare/<old_archived_service_id>...<new_archived_service_id>',
+    methods=['GET']
+)
+def compare(old_archived_service_id, new_archived_service_id):
+
+    def validate_archived_services(old_archived_service, new_archived_service):
+
+        if old_archived_service.get('id', -1) \
+                != new_archived_service.get('id', -2):
+            return False
+
+        old_updated_at = datetime.strptime(
+            old_archived_service.get('updatedAt'), DATETIME_FORMAT)
+
+        new_updated_at = datetime.strptime(
+            new_archived_service.get('updatedAt'), DATETIME_FORMAT)
+
+        if old_updated_at >= new_updated_at:
+            return False
+
+        return True
+
+    try:
+        service_data_revision_1 = data_api_client.get_archived_service(
+            old_archived_service_id)['services']
+
+        service_data_revision_2 = data_api_client.get_archived_service(
+            new_archived_service_id)['services']
+
+        # ids exist, ids match, dates are chronological
+        if not validate_archived_services(
+                service_data_revision_1, service_data_revision_2):
+            raise ValueError
+
+        service_data = data_api_client.get_service(
+            service_data_revision_1['id'])['services']
+
+    except (HTTPError, KeyError, ValueError):
+        return abort(404)
+
+    content = service_content.get_builder().filter(service_data)
```

Comment:
This will list sections that the admin user can edit, not the ones that could've been changed by the supplier. The admin manifest is probably a superset of the supplier one, so this shouldn't be an issue, but I wonder if we should make a note of it somewhere.